plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -285,20 +289,105 @@ export class EventsProcessor {
         event: string,
         properties: Properties,
         distinctId: string,
-        teamId: number
+        teamId: number,
+        timestamp: DateTime
     ): Promise<void> {
         if (isDistinctIdIllegal(distinctId)) {
             this.pluginsServer.statsd?.increment(`illegal_distinct_ids.total`, { distinctId })
             return
         }
         if (event === '$create_alias') {
-            await this.alias(properties['alias'], distinctId, teamId, false)
+            await this.merge(properties['alias'], distinctId, teamId, false, timestamp)
         } else if (event === '$identify' && properties['$anon_distinct_id']) {
-            await this.alias(properties['$anon_distinct_id'], distinctId, teamId)
+            await this.merge(properties['$anon_distinct_id'], distinctId, teamId, true, timestamp)
         }
     }
 
-    private async alias(
+    private async merge(
+        previousDistinctId: string,
+        distinctId: string,
+        teamId: number,
+        isIdentifyCall: boolean,
+        timestamp: DateTime
+    ): Promise<void> {
+        // No reason to alias person against itself. Done by posthog-js-lite when updating user properties
+        if (distinctId === previousDistinctId) {
+            return
+        }
+        if (this.isNewPersonPropertiesUpdateEnabled(teamId)) {
+            await this.mergeNew(distinctId, previousDistinctId, teamId, timestamp)
+        } else {
+            await this.aliasDeprecated(previousDistinctId, distinctId, teamId, isIdentifyCall)
+        }
+    }
+
+    private async mergeNew(
+        distinctId_1: string,
+        distinctId_2: string, // more optimal if this person has less properties compared to id_1
+        teamId: number,
+        timestamp: DateTime,
+        retriesLeft = MAX_FAILED_PERSON_MERGE_ATTEMPTS
+    ): Promise<void> {
+        let kafkaMessages: ProducerRecord[] = []
+        try {
+            await this.db.postgresTransaction(async (client) => {
+                // iff person exists, then corresponding posthog_person & posthog_persondistinctid are locked for changes
```

Comment:
```suggestion
                // if person exists, then lock the corresponding posthog_person & posthog_persondistinctid rows for changes
```

Reply:
I did intend it to be `iff` i.e. `if and only if`, was that confusing?